### PR TITLE
Coordinated forks: maxEpoch parameter and a new vote format

### PR DESCRIPTION
**Pull request by LukasGasior1**
The fork vote is no longer fully obfuscated. Thanks to that we can now store the voting result at the end of each epoch (in contrast to storing all votes as in the previous approach).

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkVote.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* Copyright 2021 Radix Publishing Ltd incorporated in Jersey (Channel Islands).
+/* Copyright 2022 Radix Publishing Ltd incorporated in Jersey (Channel Islands).
```
[the remark this answers is not in the document]
Comment (LukasGasior1) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkVote.java:1]: shouldn't we fix it on master then?

**Review by siy — APPROVED** [review]
LGTM

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkVote.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+  public static final int NONCE_HASH_LEN = 8;
+  public static final int TOTAL_LEN = NAME_LEN + FORK_CONFIG_PARAMS_HASH_LEN + NONCE_HASH_LEN;
+
+  public static CandidateForkVote create(ECPublicKey publicKey, CandidateForkConfig forkConfig) {
+    final var payload = new byte[TOTAL_LEN];
+
+    final var nameEncoded = forkConfig.name().getBytes(ForkConfig.FORK_NAME_CHARSET);
+    System.arraycopy(nameEncoded, 0, payload, 0, nameEncoded.length);
+
+    final var forkConfigParamsHash = forkConfigParamsHash(forkConfig).asBytes();
```
Comment (LukasGasior1) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/CandidateForkVote.java:91]: to be renamed to `candidateForkId` in a subsequent PR

**Comment by LukasGasior1** [comment]
superseded by https://github.com/radixdlt/radixdlt/pull/645

**Pull request closed without merging**